build-system/eslint-rules/no-style-property-setting.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const path = require('path')
+
+module.exports = function(context) {
+  const sourceCode = context.getSourceCode();
+  return {
+    Program: function(node) {
+      // Ignore style.js, fixed-layer.js and tests for usage of style property
+      // changes.
+      if (!/^(style|fixed-layer|test-(\w|-)+)\.js$/.test(
+            path.basename(context.getFilename()))) {
+        const sourceCodeAsText =
```

Comment:
The power of ESLint comes from the AST, let's take advantage of that:

```js
export default function(context) {
  return {
    MemberExpression(node) {
      // Ignore style.js, fixed-layer.js and tests for usage of style property
      // changes.
      const filename = context.getFilename();
      if (/test-/.test(filename)) {
        return;
      }
      if (/src\/(style|fixed-layer)\.js$/.test(filename)) {
        return;
      }
      if (node.computed) {
        return;
      }
      
      if (node.property.name === "style") {
        context.report(node, 'The use of Element#style (CSSStyleDeclaration live object) to style elements is forbidden. Use getStyle and setStyle from style.js');
      }
    },
  };
};
```

Reply:
ah this is more concise. I'll make these changes. Thanks Justin.

Reply:
changes made. please review. thanks